File: app/src/main/java/therajanmaurya/opensource/sweeterror/SweetMainActivity.java
package therajanmaurya.opensource.sweeterror;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;
import butterknife.OnClick;
import therajanmaurya.opensource.sweeterror.sweeterror.SweetErrorType;
import therajanmaurya.opensource.sweeterror.utils.ConstantKeys;

public class SweetMainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
    }

    @OnClick(R.id.btn_empty_ui)
    void showEmptyUI() {
        sweetIntent(SweetErrorType.EMPTY_UI);
    }

    @OnClick(R.id.btn_error_ui)
    void showErrorUI() {
        sweetIntent(SweetErrorType.ERROR_UI);
    }

    @OnClick(R.id.btn_no_internet_ui)
    void showNoInternetUI() {
        sweetIntent(SweetErrorType.NO_INTERNET);
    }

    @OnClick(R.id.btn_custom_error_ui)
    void showCustomErrorUI() {
        sweetIntent(SweetErrorType.CUSTOM);
    }

    public void sweetIntent(SweetErrorType sweetErrorType) {
        Intent sweetIntent = new Intent(this, SweetErrorInActivity.class);
        sweetIntent.putExtra(ConstantKeys.SWEET_ERROR_TYPE, sweetErrorType);
        startActivity(sweetIntent);
    }
}

File: app/src/main/java/com/github/therajanmaurya/sweeterror/demo/SweetMainActivity.kt
package com.github.therajanmaurya.sweeterror.demo

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*


class SweetMainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnEmptyUI.setOnClickListener { sweetIntent(SweetErrorType.EMPTY_UI) }
        btnErrorUI.setOnClickListener { sweetIntent(SweetErrorType.ERROR_UI) }
        btnNoInternetUI.setOnClickListener { sweetIntent(SweetErrorType.NO_INTERNET) }
        btnCustomErrorUI.setOnClickListener { sweetIntent(SweetErrorType.CUSTOM) }
        btnCustomStyleErrorUI.setOnClickListener { sweetIntent(SweetErrorType.CUSTOM_STYLE) }
    }

    private fun sweetIntent(sweetErrorType: SweetErrorType) {
        val sweetIntent = Intent(this, SweetErrorInActivity::class.java)
        sweetIntent.putExtra(ConstantKeys.SWEET_ERROR_TYPE, sweetErrorType)
        startActivity(sweetIntent)
    }
}
